ckanext/ontario_theme/plugin.py

```diff
@@ -394,6 +394,187 @@ def get_keyword_count(keyword_lang, language):
+
+    current_page
+        The page number currently being displayed.
+
+    items_per_page
+        Maximum number of items to display per page. 
+        Defined somewhere as 20.
+    '''
+    item_count =  len(all_items)
+
+    # Calculate number of pages in the pagination needed to display
+    # all items in input_items
+    if item_count > 0:
+        first_page = 1
+        page_count = int(((item_count - 1) / items_per_page) + 1)
+        last_page = first_page + page_count - 1
+
+    # Set index ranges for slicing the sorted_org object
+    slice0 = [] 
+    slice1 = []
+    for idx in range(last_page):
+        slice0.append(idx * items_per_page)
+        next_max = slice0[idx] + items_per_page
+        if item_count < next_max:
+            slice1.append(item_count)
+        else:
+            slice1.append(next_max)
+
+    # Slice the sorted_org object according to the current
+    # pagination page and return it for display
+    page = 0
+    for idx in range(len(slice0)):
+        page+=1
+        if page == current_page:
+            this_slice = all_items[slice0[idx]:slice1[idx]]
+    
+    return this_slice 
+
+def get_all_packages(**kwargs):
+    '''Helper to return the full number of packages matching a 
+    search query, including any facet search requests (in the
+    case of no searches, all packages are returned). The full
+    number of packages is needed because in the search page, 
+    only the paginated number of packages for the current page 
+    is available, which prohibits application of any custom 
+    sorting since sorting must be done on the full list before
+    pagination.
+
+    q
+        Search query. Passed through c.q.
+    
+    request_params_items
+        Contains the facet search parameters. Passed through
+        request.params.
+
+    current_org
+        When datasets are accessed by clicking on an Organization,
+        the package search must be limited to the current
+        organization, which is stored in this variable. This 
+        variable is not needed when accessing datasets through
+        the Datasets page.
+    
+    item_count
+        Number of packages matching search. Passed through 
+        c.page.item_count.
+
+    '''
+    q = kwargs['q']
+    request_params_items = kwargs['request_params_items']
+    current_org = kwargs['current_org']
+    item_count = kwargs['item_count']
+
+    # Excerpt from search() fn in ckan/ckan/controllers/package.py
+    # Needed to collect search parameters from facets.
+    search_extras = {}
+    fq = ''
+    for (param, value) in OrderedDict(request_params_items).items():
+        if param not in ['q', 'page', 'sort'] \
+            and len(value) and not param.startswith('_'):
+            if not param.startswith('ext_'):
+                fq += ' %s:"%s"' % (param, value)
+            else:
+                search_extras[param] = value
+    
+    if current_org:
+        fq += ' %s:"%s"' % ('organization', current_org)
```

Comment:
Sort resolution isn't currently applied to groups. Sorting by name ascending/descending on an individual groups page will sort the correct number of datasets but from the entire list of datasets, not the list of datasets belonging to that group.

Reply:
Should be resolved now